File: scala/a05.scala
import  scala.io.StdIn._

object Main {
    def main(args: Array[String]): Unit= {
        val Array(n, k) = readLine().split(" ").map(_.toInt)
        val count = (for {
            i <- 1 to n
            j <- 1 to n
            z = k - i - j
            if (z >= 1 && z <= n)
        } yield i).length
        println(count)
    }
}

File: cpp/a05.cpp
#include <iostream>
using namespace std;
 
int main() {
    int n, k;
    cin >> n >> k;
    int count = 0;
    for (int i = 1; i <= n; i++) {
        for (int j = 1; j <= n; j++) {
            if ( k - i - j >= 1 && k - i - j <= n) count++;
        }
    }
    cout << count;
    return 0;
}